source/Halibut/Transport/Protocol/RegisteredSerializationBinder.cs

```diff
@@ -1,109 +1,38 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Reflection;
-using Halibut.Util;
 using Newtonsoft.Json.Serialization;
 
 namespace Halibut.Transport.Protocol
 {
     public class RegisteredSerializationBinder : ISerializationBinder
     {
-        readonly Type[] protocolTypes = new[] { typeof(ResponseMessage), typeof(RequestMessage) };
-        readonly HashSet<Type> allowedTypes = new HashSet<Type>();
         readonly ISerializationBinder baseBinder = new DefaultSerializationBinder();
-        
-        public RegisteredSerializationBinder()
-        {
-            foreach (var protocolType in protocolTypes)
-            {
-                RegisterType(protocolType, protocolType.Name, true);    
-            }
-        }
-
-        public void Register(params Type[] registeredServiceTypes)
-        {
-            foreach (var serviceType in registeredServiceTypes)
-            {
-                foreach (var method in serviceType.GetHalibutServiceMethods())
-                {
-                    RegisterType(method.ReturnType, $"{serviceType.Name}.{method.Name}:{method.ReturnType.Name}", false);
-                    
-                    foreach (var param in method.GetParameters())
-                    {
-                        RegisterType(param.ParameterType,$"{serviceType.Name}.{method.Name}(){param.Name}:{param.ParameterType.Name}", false);
-                    }
-                }
-            }
-        }
-        
-        void RegisterType(Type type, string path, bool ignoreObject)
-        {
-            if (!type.AllowedOnHalibutInterface())
-            {
-                if (ignoreObject)
-                {
-                    return;
-                }
-                
-                throw new TypeNotAllowedException(type, path);
-            }
-
-            lock (allowedTypes)
-            {
-                if (!allowedTypes.Add(type))
-                {
-                    // Seen this before, no need to go further
-                    return;
-                }
-            }
+        readonly Type[] protocolTypes = { typeof(ResponseMessage), typeof(RequestMessage) };
```

Comment:
It feels like these protocol types should be part of the type registry, as it's a type concern and that's what the registry is all about.

I know registering it here means we know those types exist. That's the plus of keeping it here.

But also, it feels a little out of place that we are taking this registry, and manipulating it in the constructor. It feels like an expected consequence.